tree/dataframe/inc/ROOT/RSnapshotOptions.hxx

```diff
@@ -20,17 +20,16 @@ namespace ROOT {
 namespace RDF {
 /// A collection of options to steer the creation of the dataset on file
 struct RSnapshotOptions {
-   using ECAlgo = ::ROOT::ECompressionAlgorithm;
    RSnapshotOptions() = default;
    RSnapshotOptions(const RSnapshotOptions &) = default;
    RSnapshotOptions(RSnapshotOptions &&) = default;
-   RSnapshotOptions(std::string_view mode, ECAlgo comprAlgo, int comprLevel, int autoFlush, int splitLevel, bool lazy)
-      : fMode(mode), fCompressionAlgorithm(comprAlgo), fCompressionLevel{comprLevel}, fAutoFlush(autoFlush),
+   RSnapshotOptions(std::string_view mode, int comprAlgo, int comprLevel, int autoFlush, int splitLevel, bool lazy)
```

Comment:
doesn't this break existing user code? 

(also why int, and not `ROOT::RCompressionSetting::EAlgorithm` directly?)

Reply:
I started exactly with same approach you propose but we actually break user code in a such way:
```
root [0] ROOT::RDF::RSnapshotOptions s3("RECREATE", ROOT::kZLIB, 0, 0, 99, false)
ROOT_prompt_0:1:29: error: no matching constructor for initialization of 'ROOT::RDF::RSnapshotOptions'
ROOT::RDF::RSnapshotOptions s3("RECREATE", ROOT::kZLIB, 0, 0, 99, false)
                            ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/oksana/CERN_sources/root/builds/include/ROOT/RSnapshotOptions.hxx:26:4: note: candidate constructor not viable: no known conversion from 'ROOT::(anonymous enum at /home/oksana/CERN_sources/root/builds/include/Compression.h:87:1)' to 'ROOT::RCompressionSetting::EAlgorithm' for 2nd argument
   RSnapshotOptions(std::string_view mode, ROOT::RCompressionSetting::EAlgorithm comprAlgo, int comprLevel, int autoFlush, int splitLevel, bool lazy)
   ^
```
So the problem is that ECAlgo was before enum, but now ROOT::RCompressionSetting::EAlgorithm is enum class.(while ROOT::RCompressionSetting is struct). These PR is just one step closer to last transition step, when we will be remove all magic ints from ROOT codebase. Because of this reason I should introduce some static_casts to not to break a user code.. Currently with my changes dataframe_ranges test is compiling and working as expected..

Reply:
maybe I'm missing something: you had to change `dataframe_snapshot.cxx` to make this work, so similarly users will have to change their code right? (i.e. this will break user code)

Reply:
>  but now ROOT::RCompressionSetting::EAlgorithm is enum class.

I thought we agreed not to make it a enum class in order to be backward compatible with old user code.

Reply:
@pcanal, it didn't work out, I will try one more iteration to converge without breaking user code in RDF,  for other cases we are covered.

Reply:
@bluehood @pcanal I left everything without changes, so user code is not broken. On this stage, looks like I can't do anything better.